modules/container_definition/secrets.tf

```diff
@@ -0,0 +1,15 @@
+locals {
+  ssm_prefix = "/aws/reference/secretsmanager/"
+
+  secrets_keys        = keys(var.secrets)
+  sorted_secrets_keys = sort(local.secrets_keys)
+  sorted_secrets = [
+    for key in local.sorted_secrets_keys :
+    {
+      name      = key
+      valueFrom = "${local.ssm_prefix}${lookup(var.secrets, key)}"
+    }
+  ]
+
+  final_secrets = length(local.sorted_secrets) > 0 ? local.sorted_secrets : null
```

Comment:
See other comment

Reply:
I think this may not be what we want in this case (this returns `null` rather than `[]` so the attribute will be filtered out if empty - I will check whether secrets needs to be `null` or []` in the container definition to prevent churn.

Reply:
It does not cause churn, so will fix.